### ISSUE TITLE
您这个架构的一个目的是保持Activity或Fragment中不出现DataBinding中的内容，但是如果要配置一些组件的程序，比如viewPager2要使用TabLayoutMediator，需要进行acctch() 和detach()的调用，也就是要绑定周明周期，这种情况需要怎么处理呢

### ISSUE THREAD

**Issue by zion223** [KunMinX/Jetpack-MVVM-Best-Practice#39]
您这个架构的一个目的是保持Activity或Fragment中不出现DataBinding中的内容，但是如果要配置一些组件的程序，比如viewPager2要使用TabLayoutMediator，需要进行acctch() 和detach()的调用，也就是要绑定周明周期，这种情况需要怎么处理呢

**Comment by zion223** [comment]
@Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
//databing 
        viewPager2.registerOnPageChangeCallback(mPageChangeCallback);
        mediator = new TabLayoutMediator(tabLayout, viewPager2, true, new TabLayoutMediator.TabConfigurationStrategy() {
            @Override
            public void onConfigureTab(@NonNull TabLayout.Tab tab, int position) {
                tab.setCustomView(makeTabView(position));
            }
        });
        mediator.attach();
}


    @Override
    public void onDestroy() {
        mediator.detach();
        viewPager2.unregisterOnPageChangeCallback(mPageChangeCallback);
        super.onDestroy();
    }

**Comment by KunMinX** [comment]
@zion223 

可以将这些内容封装到 BindingAdapter 中，并通过绑定 ViewModel 中的数据来驱动这些方法的执行，具体可参考 TabPageBindingAdapter

**Comment by zion223** [comment]
感谢解答
